File: 1894-merge-strings-alternately/1894-merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let indexA = 0
    let indexB = 0
    let str = []
    while (indexA < word1.length || indexB < word2.length) {
        if (indexA <= indexB) {
            str.push(word1.charAt(indexA))
            indexA++;
        } else {
            str.push(word2.charAt(indexB))
            indexB++;
        }
    }

    if (indexA >= word1.length) {
        str.push(word2.substring(indexA).split(""))
    }
    // if (indexB >= word2.length) {
    //     // const temp = word1.substring(indexB).split("")
    //     // console.log(temp)
    //     // str.push(word1.substring(indexB).split(""))
    // } 

    // console.log(indexA,indexB, str)
    return String(str.join(""));
};

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder ret = new StringBuilder();
        int leftIndex = 0;
        int rightIndex = 0;

        while (leftIndex < word1.length() && rightIndex < word2.length()) {
            ret.append(word1.charAt(leftIndex));
            ret.append(word2.charAt(rightIndex));
            leftIndex++;
            rightIndex++;
        }

        if (word1.length() > word2.length()) {
            ret.append(word1.substring(leftIndex));
        } else {
            ret.append(word2.substring(rightIndex));
        }

        return ret.toString();
    }
}